File: 1370-increasing-decreasing-string/1370-increasing-decreasing-string.py
class Solution:
    def sortString(self, s: str) -> str:
        arr, res = [0] * 26, ""
        for c in s:
            arr[ord(c)-97] += 1
        while sum(arr) > 0:
            for i in range(0, len(arr)):
                if arr[i] > 0:
                    arr[i] -= 1
                    res += chr(i+97)
                    break
            for i in range(ord(res[-1])-97+1, len(arr)):
                if arr[i] > 0:
                    arr[i] -= 1
                    res += chr(i+97)
            for i in range(len(arr)-1, -1, -1):
                if arr[i] > 0:
                    arr[i] -= 1
                    res += chr(i+97)
                    break
            for i in range(ord(res[-1])-97-1, -1, -1):
                if arr[i] > 0:
                    arr[i] -= 1
                    res += chr(i+97)
        return res

File: 1370-increasing-decreasing-string/1370-increasing-decreasing-string.ts
function sortString(s: string): string {
    const arr = Array<number>(26).fill(0)
    let res = ""
    for (const c of s) {
        arr[c.charCodeAt(0)-97]++
    }
    while (arr.some(Boolean)) {
        for (let i=0; i < arr.length; i++) {
            if (arr[i] > 0) {
                res += String.fromCharCode(i+97)
                arr[i]--
                break
            }
        }
        for (let i=Math.min(res.at(-1).charCodeAt(0)-97+1, arr.length-1); i < arr.length; i++) {
            if (arr[i] > 0) {
                res += String.fromCharCode(i+97)
                arr[i]--
            }
        }
        for (let i=arr.length-1; i >= 0; i--) {
            if (arr[i] > 0) {
                res += String.fromCharCode(i+97)
                arr[i]--
                break
            }
        }
        for (let i=Math.max(res.at(-1).charCodeAt(0)-97-1, 0); i >= 0; i--) {
            if (arr[i] > 0) {
                res += String.fromCharCode(i+97)
                arr[i]--
            }
        }
    }
    return res
};